components/script/dom/url.rs

```diff
@@ -24,19 +26,24 @@ pub struct URL {
 
     // https://url.spec.whatwg.org/#concept-urlutils-get-the-base
     base: Option<Url>,
+
+    // https://url.spec.whatwg.org/#dom-url-searchparams
+    query: JS<URLSearchParams>,
```

Comment:
Firefox iterates index-wise, if you change the query string it will continue to iterate from that index in the new query string if that index exists.

```js
let a=new URL("http://a.b/c?a=1&b=2&c=3&d=4");
let b = a.searchParams;
for (i of b) {
    a.search="x=1&y=2&z=3";
    console.log(i);
}
// outputs
Array [ "a", "1" ]
Array [ "y", "2" ]
Array [ "z", "3" ]
```